### DISCUSSION TITLE
Filament completely unloads at start of print?

### DISCUSSION THREAD

**Discussion by cds333** [mriscoc/Ender3V2S1#526] [Q&A]
It has happened on multiple E3V2s, one with MM and one with ABL, running the respective firmwares. I believe it happened on both the 4.2.2 and 4.2.7. The extruder just unloads the filament completely as soon as the print starts. 

Running Cura's stock startup gcode for the E3V2 with M420 added.

I printed the exact same file on the same printer later or earlier the same day and it did not do the unload.

It has happened a dozen or more times now, seemingly randomly.

Is this a known issue?

**Comment by mriscoc** [comment]
Check your filament runout sensor connection or configuration.

**Reply by cds333** [reply]
Neither printer has a runout sensor. Which is why the randomness of it is so confusing.

So I should just disable runout sensor in the firmware before compiling?

**Reply by mriscoc** [reply]
These MM versions work fine without runout sensors

https://github.com/mriscoc/Ender3V2S1/releases/download/20221002/Ender3V2-422-MM-20221002.bin
https://github.com/mriscoc/Ender3V2S1/releases/download/20221002/Ender3V2-427-MM-20221002.bin

Also runout sensor can be disabled by menu or in the start G-code script [M412 S0](https://marlinfw.org/docs/gcode/M412.html)

**Reply by cds333** [reply]
They are mostly ABL enabled firmware versions fyi. Only did one MM recently. 

OK I found the setting, I have disabled it and will report back if it happens again. Hopefully that will do it, thank you!

Any idea what could be causing this? Is it a bug in the firmware or a bug on the board? A floating ground perhaps on the runout sensor pins? 

I would like to get this solved; I install your firmware on every E3 I come across and tell everyone it is the best available, and try to steer than away from Jyers, but this problem causes inevitable irritation.

Do we know if the runout sensor pins are NO or NC? Could I just install a jumper perhaps?

**Reply by mriscoc** [reply]
With the provided configuration files for own compilation it must works without problems.

https://github.com/mriscoc/Special_Configurations

**Comment by MarkforU** [comment]
I'm running V2 with 4.2.2 and 4.2.7 boards, left filament runout at stock firmware settings, basically because I forgot to check them when upgrading firmware, both of my systems do filament runout just fine. I remember I had them set at low, on old firmware that worked. I'm printing several day prints right now, so I can't check settings, but like I said after updating, I forgot to set them. I'm running the $12 filament runout sensor that comes with bracket...since I forgot to set my filament runout, I was ready to hit the button when my filament ran out, but it took care of itself without any problems. Either your sensor or settings...

**Reply by cds333** [reply]
Neither printer has a sensor.
